### PR TITLE
Us123549/user selector list

### PR DESCRIPTION
**Pull request by johngwilkinson**
Quality Notes
- functional (e2e w. local LMS containing related API changes)
  - regression: view Engagement Dashboard as admin and as auditor
  - from insights portal, click on new tile and confirm list loads
    - as admin
    - as auditor
  - click on a user and confirm we see their user drill page
    - as admin
    - as auditor
  - navigation
    - correct back text shown in user drill and user selection pages
    - use back to go back and select a different user, and to go back to Insights Portal
  - edge case: auditee with no courses
    - user drill loads and shows standard "no courses in selection" message
  - [x] profile pictures (had it working before; currently can't get actual images to appear on this page or the existing user drill one)
- security: n/a
- performance
  - after discussion with UX, will be using a Load More design instead of paging; this should be easy to get performant
  - server returns just a few search results (e.g. 10) each time
  - profile pics load relatively slowly, but this doesn't stop the user from using the page in the meantime
- devops: currently feature-flagged off in the LMS
- UX/docs
  - arabic renders fine (RTL)
  - keyboard/screenreader navigation
    - note: reader reads the default profile images (the user's initials), which is mildly annoying, but a feature of the standard d2l component we are using

FYI @Vitalii-Misechko @anhill-D2L 

image

### CODE REVIEW

File: components/immersive-nav.js [hunk 1]
```diff
@@ -70,6 +70,8 @@ class InsightsImmersiveNav extends Localizer(MobxLitElement) {
 			case 'home':
 				return this.localize('dashboard:backToInsightsPortal');
 			case 'user':
+				if (this.viewState.isSingleLearner) return 'Select a Different User'; // TODO: localize
```
Comment (johngwilkinson) [inline components/immersive-nav.js:73]: TODO

**Comment by johngwilkinson** [comment]
Update on profile pictures: there's a perm under User Profiles that is needed see others' profiles. Administrator does not have it by default. With a couple of pics:

image

**Comment by NicholasHallman** [comment]
Took a quick look, the user profiles don't seem to have skeletons when loading. Is this just me?

**Comment by johngwilkinson** [comment]
> Took a quick look, the user profiles don't seem to have skeletons when loading. Is this just me?

I think there are two things here.

One I forgot to mention: it was trying to load actual images for the skeleton, so 10 HTTP 404s would should up in the console. So I put in the code that omits the entity ref if skeleton rows - but that doesn't show the user profiles while the list skeleton is visible. So far, I haven't found a way to have a skeleton image that doesn't throw errors, but I'm sure it's possible.

The second thing is after the user list has loaded, you're right that there are no skeletons for the user images while the actual images load. I don't know if we need that or not, but it's true of the user drill page too.

**Comment by NicholasHallman** [comment]
> I don't know if we need that or not, but it's true of the user drill page too.

Oh yeah good point, instead of showing the blank profile icon I added the d2l-skeletize class to the profile in the user drill.

skeletize user profile

**Review by NicholasHallman — APPROVED** [review]

File: model/selectorFilters.js [hunk 2]
```diff
@@ -128,6 +128,33 @@ export class OrgUnitSelectorFilter {
...
 
+export class UserSelectorFilter {
+	constructor(data) {
+		this.selected = data.serverData.selectedUserId;
+		// noinspection JSUnusedGlobalSymbols
+		this._urlState = new UrlState(this);
+	}
+
+	// persistence key and value for UrlState
+	get persistenceKey() { return 'lf'; }
```
[the remark this answers is not in the document]
Comment (johngwilkinson) [reply model/selectorFilters.js:139]: "Learner Filter"

**Comment by johngwilkinson** [comment]
@Vitalii-Misechko I think this PR is blocked on the same sauce issue you're looking at.

**Comment by Vitalii-Misechko** [comment]
> @Vitalii-Misechko I think this PR is blocked on the same sauce issue you're looking at.

That's correct. I've figured out that `dequelabs/axe-core` released `4.2.0` 2 days ago.  Souce labs installs this version of the dependency, therefore, we see these failing tests. I created [PR ](https://github.com/Brightspace/insights-engagement-dashboard/pull/553)that temporarily excludes these test until I fix them properly

**Comment by Vitalii-Misechko** [comment]
Re-launched CI. Should be green now

**Pull request merged**